In [52]:
import pandas as pd
import numpy as np
from scipy import stats
from statsmodels.stats.proportion import proportions_ztest

# __Problem Statement :__  

_Fantaloons Sales managers commented that % of males versus females walking in to the store differ based on day of the week. Analyze the data and determine whether there is evidence at 5 % significance level to support this hypothesiss_

## __1 - Business Problem__  
Does the % of males versus females walking in to the store differ based on day of the week ?

## __2 - Data description__ 

α == 0.05 (95% Confidence)

Y == Discrete  
X == Discrete  

Since there are 2 variable we will perform 2 proportion test

## __3 - 2 Proportion Test__  

In [14]:
df = pd.read_csv("C:/Users/Sohail Shaikh/Downloads/Faltoons.csv")

H0 == The % of males versus females walking in to the store based on day of the week is equal  
H1 == The % of males versus females walking in to the store based on day of the week is not equal

In [32]:
df.Weekend.str.count("Male").sum() # 167

167

In [33]:
df.Weekend.str.count("Female").sum() # 233

233

In [34]:
df.Weekdays.str.count("Male").sum() # 133

113

In [35]:
df.Weekdays.str.count("Female").sum() # 287

287

In [59]:
female_count = np.array([287, 233])

In [60]:
total_count = np.array([420,400])

In [61]:
statsmodels.stats.proportion.proportions_ztest(female_count, total_count)

(2.9964248640358, 0.0027316554639882896)

(2.9964248640358, 0.0027316554639882896)  
P value of 2 Propotion test  == 0.0027 < α

## Thus H1 is accepted.  
## The % of males versus females walking in to the store based on day of the week is not same

H0 == The % of females walking in to the store based on day of the week > then % of males  
H1 == The % of males walking in to the store based on day of the week < then % of males

In [62]:
statsmodels.stats.proportion.proportions_ztest(female_count, total_count, alternative = 'smaller')

(2.9964248640358, 0.9986341722680059)

(2.9964248640358, 0.9986341722680059)  
P value of 2 Propotion Test == 0.99 > α  

## Thus H0 is accepted.  
## The % of females walking in to the store based on day of the week > then % of males